### ISSUE TITLE
Developing a custom UserRegistry

### ISSUE THREAD

**Issue by ggam** [OpenLiberty/docs#2284]
I tried to develop a custom `UserRegistry` but I wasn't able to do it.

There are some articles on ibm.com (https://www.ibm.com/support/knowledgecenter/SS7K4U_liberty/com.ibm.websphere.wlp.zseries.doc/ae/twlp_sec_custmr.html) but they mess a lot with OSGi and I couldn't make it work. A new guide for this topic would be greatly appreciated.

**Comment by dmuelle** [comment]
Thanks for your feedback @ggam, we are looking into this. I'm researching an alternative solution but waiting on testing.

**Comment by dmuelle** [comment]
closing this issue, work for a new task topic on this process is being tracked in #2418

**Comment by lauracowen** [comment]
@ggam Can I ask why you've chosen to write a custom user registry rather than using an existing one in some form? Or did you mean you already have a custom user registry that you want to configure to work with your app on Liberty?

**Comment by ggam** [comment]
@lauracowen I need to get users from a database and authenticate via both MP JWT or basic auth. I haven't found a way to get users from the DB with the default registries (only LDAP).

I can use Jakarta Security to get the users from the database, but Open Liberty disables MP JWT when Jakarta Security is in place.

**Comment by Emily-Jiang** [comment]
@ggam I am surprised that your obeservation of "Open Liberty disables MP JWT when Jakarta Security is in place.", which should not be the expected behaviour. When you say Jakarta Security, do you mean Jakarta EE Security such as `@RolesAllowed` etc? If yes, it should not be case as MP JWT has to rely on that to secure applications. If no, can you explain with more details. If you have github repo to demonstrate this, it will be very much appreciated! Thanks you for reporting this issue!

**Comment by Emily-Jiang** [comment]
reopened this for further clarification

**Comment by Emily-Jiang** [comment]
@ggam we are going to close this issue since there is no response from you for nearly a month. Please reopen with the response for the questions listed in [this](https://github.com/OpenLiberty/docs/issues/2284#issuecomment-724738217) comment.

**Comment by ggam** [comment]
Hi Emily and sorry for the late reply. I meant former JSR 375. I opened an
issue but was told it is working as designed:
https://github.com/OpenLiberty/open-liberty/issues/12253

**Comment by c-koell** [comment]
@Emily-Jiang I think @ggam described the same issues as we had also before. https://github.com/OpenLiberty/open-liberty/issues/11058
If the HAM is activated the mpJwt Security will be ignored.
